bayesian/api_v1.py

```diff
@@ -347,35 +347,42 @@ def get(external_request_id):
                 started_at = stack_result["task_result"]["_audit"]["started_at"]
                 finished_at = stack_result["task_result"]["_audit"]["ended_at"]
 
-
         if reco_result is not None and 'task_result' in reco_result:
-            if reco_result["task_result"] != None:
-                recommendation = reco_result['task_result']['recommendations']
+                if reco_result["task_result"] != None:
+                    recommendation = reco_result['task_result']['recommendations']
 
         # Populate sentiment score for packages in user's stack
         if stack_result is not None:
-            user_stack_deps = stack_result['task_result']['user_stack_info']['dependencies']
-            for dep in user_stack_deps:
-                if user_stack_sentiment_result is not None:
-                    dep['sentiment']['overall_score'] = user_stack_sentiment_result['task_result'][dep['name']]['score']
-                else:
-                    dep['sentiment'] = {}
+            user_stack_deps = stack_result.get('task_result').get('user_stack_info').get('dependencies',[])
+            if user_stack_deps:
+                for dep in user_stack_deps:
+                    if user_stack_sentiment_result is not None:
+                        if user_stack_sentiment_result.get('task_result').get(dep['name']) is not None:
+                            dep['sentiment']['overall_score'] = user_stack_sentiment_result.get('task_result').\
+                                get(dep['name']).get('score', 0)
+                        else:
+                            dep['sentiment'] = {}
 
         # Populate sentiment score for recommended packages
         if reco_result is not None:
-            alternate = reco_result['task_result']['recommendations']['alternate']
-            for pkg in alternate:
-                if reco_pkg_sentiment_result is not None:
-                    pkg['sentiment']['overall_score'] = reco_pkg_sentiment_result['task_result'][pkg['name']]['score']
-                else:
-                    pkg['sentiment'] = {}
+            alternate = reco_result.get('task_result').get('recommendations').get('alternate', [])
+            if alternate:
+                for pkg in alternate:
+                    if reco_pkg_sentiment_result.get("task_result") is not None:
```

Comment:
`reco_pkg_sentiment_result` might be None ... we need to check for that just like you checked for `user_stack_sentiment_result`

Also, value of `reco_pkg_sentiment_result.get('task_result')` can be kept in a new variable e.g. `reco_pkg_sentiment` so that we can avoid fetching the `task_result` value again and again. This is also more readable.